kroxylicious-authorizer-providers/kroxylicious-authorizer-acl/src/main/java/io/kroxylicious/authorizer/provider/acl/AclAuthorizerService.java

```diff
@@ -0,0 +1,463 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.authorizer.provider.acl;
+
+import java.io.UncheckedIOException;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.EnumSet;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.antlr.v4.runtime.BaseErrorListener;
+import org.antlr.v4.runtime.CharStream;
+import org.antlr.v4.runtime.CharStreams;
+import org.antlr.v4.runtime.CommonTokenStream;
+import org.antlr.v4.runtime.RecognitionException;
+import org.antlr.v4.runtime.Recognizer;
+import org.antlr.v4.runtime.Token;
+import org.antlr.v4.runtime.tree.ParseTreeWalker;
+import org.antlr.v4.runtime.tree.TerminalNode;
+
+import com.google.re2j.Pattern;
+import com.google.re2j.PatternSyntaxException;
+
+import io.kroxylicious.authorizer.provider.acl.parser.AclRulesBaseListener;
+import io.kroxylicious.authorizer.provider.acl.parser.AclRulesLexer;
+import io.kroxylicious.authorizer.provider.acl.parser.AclRulesParser;
+import io.kroxylicious.authorizer.service.Authorizer;
+import io.kroxylicious.authorizer.service.AuthorizerService;
+import io.kroxylicious.authorizer.service.ResourceType;
+import io.kroxylicious.proxy.authentication.Principal;
+import io.kroxylicious.proxy.plugin.Plugin;
+import io.kroxylicious.proxy.plugin.Plugins;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+
+@Plugin(configType = AclAuthorizerConfig.class)
+public class AclAuthorizerService implements AuthorizerService<AclAuthorizerConfig> {
+
+    private @Nullable AclAuthorizer aclAuthorizer;
+
+    @Override
+    public void initialize(@Nullable AclAuthorizerConfig config1) {
+        var config = Plugins.requireConfig(this, config1);
+        var fileName = config.aclFile();
```

Comment:
Related to the point about the inability to combine files mentioned else where. 

Should we support multiple rules files? I'm thinking about a platform team with multiple tenants. They want to have one central rules file to provide basic access control and then allow the "tenants" (for want of a better term) to provide their own rules in a file per tenant format. 

I don't think its something that has to be addressed in this PR but mentioning it now while I think about it.

Reply:
The rules currently look amenable to using multiple files in future. In theory you can make a union of the denial rules, a union of the approval rules and it should work.

Definitely something to defer though since each file can alias the Principals how they like you can't just munge two rules files together before parsing. Maybe it's more like you should be able to instantiate multiple AclAuthorizers and then smush them together into a single one.